src/sampling.jl

```diff
@@ -904,6 +906,8 @@ function sample!(rng::AbstractRNG, a::AbstractArray, wv::AbstractWeights, x::Abs
         throw(ArgumentError("output array a must not share memory with weights array wv"))
     1 == firstindex(a) == firstindex(wv) == firstindex(x) ||
         throw(ArgumentError("non 1-based arrays are not supported"))
+    isfinite(sum(wv)) || 
```

Comment:
check if `sum(wv)` can be cached (I have not read the logic enough to do it; maybe we need to pass the sum to internal functions).

Reply:
`AbstractWeights` objects always return the cached sum when `sum` is called.

Reply:
Ah - then it is OK. This should be documented then. It is not now.

Reply:
As an other thought. Maybe then we should not make this change in this PR, but instead require in types implementing `AbstractWeights` constructor that they do this check instead? It would "fix the problem at its root", and we would not have to add this check in every function that uses `AbstractWeights`?